### ISSUE TITLE
(.18) silent crash during EWBF benchmark

### ISSUE THREAD

**Issue by malventano** [DillonN/NiceHashMiner#10]
Not sure what the change to speed up the EWBF benchmark did, but it appears to be silently killing NHM .18 during the benchmark process. Appears to happen randomly. Confirmed under Windows 10 and 8.1 and on different NV GPUs (1080Ti, 1080).

**Comment by DillonN** [comment]
Does it only happen on the EWBF bench? Can you post your logs?

I'll try to reproduce on my 1080 Ti machine

**Comment by DillonN** [comment]
Can't get it to reproduce

I have a suspicion of what it might be though. The benchmarker uses the log file from the miner to get the speed, but the miner doesn't let go of the log file for around a second and reading it can cause an access denied error. I put a value that was high in all my testing but maybe not high enough. Tomorrow I'll update with logic to wait until EWBF lets go of the log

**Comment by malventano** [comment]
When the crash happens, it's only during the EWBF bench, and if I start with a cleared log folder, only the main log file will be created (no Log_GPU files). I had a 1080 crash twice in a row. Both log files are attached.

[EWBF crash logs.zip](https://github.com/DillonN/NiceHashMiner/files/1090741/EWBF.crash.logs.zip)

**Comment by DillonN** [comment]
Unfortunately the logs don't show the exception

Check tomorrow, I will patch what I think the problem is. For now you can run EWBF by manually setting the speed in settings

**Comment by malventano** [comment]
I've also noticed that all benchmark progress is lost when this crash happens. I started watching it closely and if it makes it past EWFB, I stop and restart the benchmarks so that the completed ones are saved.

**Comment by DillonN** [comment]
Yeah that's probably a byproduct of the crash, not giving NHM a chance to save previous benchmarks

**Comment by Yaro96** [comment]
Can confirm, crashing during benchmark also on GTX 1070.

**Comment by Yaro96** [comment]
I managed to fix it just by doing a clean install, on the first time I copied my old "configs/General.json" file to the root before initializating it.

**Comment by DillonN** [comment]
@Yaro96  glad you got it working!

I was still unable to get the issue to reproduce, but I'm 90% sure it was due to the bug I mentioned above (EWBF locking the log files). Now NHM will gracefully handle the error and recheck every second for the log files to be unlocked. If after 10 seconds they are still locked, it will terminate the benchmark

https://github.com/DillonN/NiceHashMiner/releases/tag/1.7.5.19

Let me know if you still have a problem

**Comment by darrenxyli** [comment]
still there

**Comment by CreamyG31337** [comment]
is it possible to change the benchmark to show the window so we can see what's happening without having to wait for it to fail?

**Comment by DillonN** [comment]
Yes I'm planning on making improvements to the benchmarking system soon

For now you can follow the [manual benchmarking](https://github.com/DillonN/NiceHashMiner/wiki/Getting-started#manualbenchmarking) instructions. If you see any errors in the EWBF window please let me know what they are.

**Comment by Yaro96** [comment]
Yes, I wasn't able to run EWBF yesterday. Here's the log:
[log.txt](https://github.com/DillonN/NiceHashMiner/files/1097085/log.txt)
